Both new Windows failures were hidden:

```
 ======================================================================
FAIL: test_shared_memory_basics (test.test_multiprocessing_spawn.WithProcessesTestSharedMemory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\cpython\cpython\lib\test\_test_multiprocessing.py", line 3794, in test_shared_memory_basics
    self.assertEqual(sms.size, sms2.size)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 512 != 4096

======================================================================
FAIL: test_checksum_fodder (test.test_tools.test_md5sum.MD5SumTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\cpython\cpython\lib\test\test_tools\test_md5sum.py", line 41, in test_checksum_fodder
    self.assertIn(part.encode(), out)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: b'@test_1772_tmp\xc3\xa6' not found in b'd38dae2eb1ab346a292ef6850f9e1a0d @test_1772_tmp\xe6\\md5sum.fodder\r\n'
```

What should I do with them? Ignore them for now and create new bug reports?

--- a/Lib/test/_test_multiprocessing.py
+++ b/Lib/test/_test_multiprocessing.py
@@ -3773,6 +3773,7 @@ class _TestSharedMemory(BaseTestCase):
         local_sms.buf[:len(binary_data)] = binary_data
         local_sms.close()
 
+    @unittest.skipIf(sys.platform == "win32", "test is broken on Windows")
     def test_shared_memory_basics(self):
         sms = shared_memory.SharedMemory('test01_tsmb', create=True, size=512)
         self.addCleanup(sms.unlink)

--- a/Lib/test/support/hashlib_helper.py
+++ b/Lib/test/support/hashlib_helper.py
@@ -22,19 +22,6 @@ def requires_hashdigest(digestname, openssl=None, usedforsecurity=True):
     ValueError: unsupported hash type md4
     """
     def decorator(func_or_class):
-        @functools.wraps(func_or_class)
-        def wrapper(*args, **kwargs):
-            try:
-                if openssl and _hashlib is not None:
-                    _hashlib.new(digestname, usedforsecurity=usedforsecurity)
-                else:
-                    hashlib.new(digestname, usedforsecurity=usedforsecurity)
-            except ValueError:
-                raise unittest.SkipTest(
-                    f"hash digest '{digestname}' is not available."
-                )
-            return func_or_class(*args, **kwargs)
-
         if isinstance(func_or_class, type):
             setUpClass = func_or_class.__dict__.get('setUpClass')
             if setUpClass is None:
@@ -47,5 +34,18 @@ def requires_hashdigest(digestname, openssl=None, usedforsecurity=True):
             setUpClass = classmethod(decorator(setUpClass))
             func_or_class.setUpClass = setUpClass
             return func_or_class
+
+        @functools.wraps(func_or_class)
+        def wrapper(*args, **kwargs):
+            try:
+                if openssl and _hashlib is not None:
+                    _hashlib.new(digestname, usedforsecurity=usedforsecurity)
+                else:
+                    hashlib.new(digestname, usedforsecurity=usedforsecurity)
+            except ValueError:
+                raise unittest.SkipTest(
+                    f"hash digest '{digestname}' is not available."
+                )
+            return func_or_class(*args, **kwargs)
         return wrapper
     return decorator

--- a/Lib/test/support/hashlib_helper.py
+++ b/Lib/test/support/hashlib_helper.py
@@ -21,30 +21,31 @@ def requires_hashdigest(digestname, openssl=None, usedforsecurity=True):
     ValueError: [digital envelope routines: EVP_DigestInit_ex] disabled for FIPS
     ValueError: unsupported hash type md4
     """
-    try:
-        if openssl and _hashlib is not None:
-            _hashlib.new(digestname, usedforsecurity=usedforsecurity)
-        else:
-            hashlib.new(digestname, usedforsecurity=usedforsecurity)
-    except ValueError:
-        should_be_skipped = True
-    else:
-        should_be_skipped = False
-
-    def maybe_skip():
-        if should_be_skipped:
-            raise unittest.SkipTest(
-                f"hash digest '{digestname}' is not available."
-            )
-
     def decorator(func_or_class):
-        if isinstance(func_or_class, type):  # The whole `class`
-            maybe_skip()
-            return func_or_class
-
-        @functools.wraps(func_or_class)  # Just a single function
+        @functools.wraps(func_or_class)
         def wrapper(*args, **kwargs):
-            maybe_skip()
+            try:
+                if openssl and _hashlib is not None:
+                    _hashlib.new(digestname, usedforsecurity=usedforsecurity)
+                else:
+                    hashlib.new(digestname, usedforsecurity=usedforsecurity)
+            except ValueError:
+                raise unittest.SkipTest(
+                    f"hash digest '{digestname}' is not available."
+                )
             return func_or_class(*args, **kwargs)
+
+        if isinstance(func_or_class, type):
+            setUpClass = func_or_class.__dict__.get('setUpClass')
+            if setUpClass is None:
+                def setUpClass(cls):
+                    super(func_or_class, cls).setUpClass()
+                setUpClass.__qualname__ = func_or_class.__qualname__ + '.setUpClass'
+                setUpClass.__module__ = func_or_class.__module__
+            else:
+                setUpClass = setUpClass.__func__
+            setUpClass = classmethod(decorator(setUpClass))
+            func_or_class.setUpClass = setUpClass
+            return func_or_class
         return wrapper
     return decorator

--- a/Lib/test/test_tools/test_md5sum.py
+++ b/Lib/test/test_tools/test_md5sum.py
@@ -1,5 +1,6 @@
 """Tests for the md5sum script in the Tools directory."""
 
+import sys
 import os
 import unittest
 from test.support import os_helper

--- a/Lib/test/test_tools/test_md5sum.py
+++ b/Lib/test/test_tools/test_md5sum.py
@@ -33,6 +33,7 @@ class MD5SumTests(unittest.TestCase):
             out.startswith(b'd41d8cd98f00b204e9800998ecf8427e <stdin>'))
         self.assertFalse(err)
 
+    @unittest.skipIf(sys.platform == "win32", "test is broken on Windows")
     def test_checksum_fodder(self):
         rc, out, err = assert_python_ok(self.script, self.fodder)
         self.assertEqual(rc, 0)

